lib/fauxpaas/lazy/directory.rb

```diff
@@ -0,0 +1,87 @@
+require "fauxpaas/lazy/file"
+require "find"
+require "pathname"
+
+module Fauxpaas
+  module Lazy
+
+    class Directory
+      def self.for(base_path, files = nil)
+        base_path = Pathname.new(base_path)
+        files ||= Find.find(base_path)
```

Comment:
So by default, a `Lazy::Directory` contains all the files at any depth under it? That seems like the correct semantics for what we're trying to do, but it's a very different notion from a standard ruby (or posix) `Directory`. Regardless, this class definitely needs some documentation.

Reply:
Class and method comments added.